From leoli@freescale.com  Tue Feb 17 17:36:15 2009
From: Li Yang <leoli@freescale.com>
Date: Fri, 13 Feb 2009 16:14:39 +0800
Subject: USB: fsl_usb2_udc: fix potential queue head corruption
To: greg@kroah.com
Cc: dbrownell@users.sourceforge.net, Li Yang <leoli@freescale.com>
Message-ID: <1234512879-31672-1-git-send-email-leoli@freescale.com>


Clear next TD field and status field in queue head initialization code
to prevent unpredictable result caused by residue of usb reset.

Signed-off-by: Li Yang <leoli@freescale.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/fsl_usb2_udc.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/gadget/fsl_usb2_udc.c
+++ b/drivers/usb/gadget/fsl_usb2_udc.c
@@ -404,7 +404,10 @@ static void struct_ep_qh_setup(struct fs
 	}
 	if (zlt)
 		tmp |= EP_QUEUE_HEAD_ZLT_SEL;
+
 	p_QH->max_pkt_length = cpu_to_le32(tmp);
+	p_QH->next_dtd_ptr = 1;
+	p_QH->size_ioc_int_sts = 0;
 
 	return;
 }
